apps/designer/app/shared/fetcher/persistent-fetcher.tsx

```diff
@@ -0,0 +1,82 @@
+import { type FetcherWithComponents, useFetcher } from "@remix-run/react";
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useRef,
+  type ReactNode,
+} from "react";
+import { useOnFetchEnd } from "./on-fetch-end";
+
+/* This utility is designed for the case when you want to send a request using Remix's fetcher,
+ * but you don't want the request to be cancelled when the component where useFetcher() is called unmounts.
```

Comment:
I am actually surprised that remix cancels the fetch when component unmounts

Reply:
I don't know 🤷‍♂️

Reply:
lets dig into this, we need to know

Reply:
I can create an issue and ask. Couldn't find anything in existing issues or docs. I the source code it looks like this:

https://github.com/remix-run/remix/blob/471ab259fba5adbdd70de71ee7bc7e874c1f4db9/packages/remix-react/components.tsx#L1469-L1474

https://github.com/remix-run/remix/blob/471ab259fba5adbdd70de71ee7bc7e874c1f4db9/packages/remix-react/transition.ts#L496-L498

Reply:
Maybe look in remixes discussions/issues or start your own. Basically we need to figure out what's remixes plan for fetching data like you did in persistent fetcher

Reply:
Ah, didn't check discussions. It was asked there before, but there're only answers by community members:

https://github.com/remix-run/remix/discussions/3585 — suggested to not unmount until done
https://github.com/remix-run/remix/discussions/2623#discussioncomment-2508116 — suggested to use context like I did
https://github.com/remix-run/remix/discussions/2462#discussioncomment-2423735 — suggested to not unmount until done

Not sure if there's a point in trying to ask again.

Reply:
well that's on itself is a good answer, so we are on the right pass, since we don't want to force user to wait, then we have to use context